src/main/java/com/yelp/nrtsearch/server/grpc/LuceneServer.java

```diff
@@ -1405,6 +1406,41 @@ public void backupWarmingQueries(
                   .asRuntimeException());
           return;
         }
+        int numQueriesThreshold = request.getNumQueriesThreshold();
+        int numWarmingRequests = warmer.getNumWarmingRequests();
+        if (numQueriesThreshold > 0 && numWarmingRequests < numQueriesThreshold) {
+          logger.warn(
+              "Unable to backup warming queries since warmer has {} requests, which is less than threshold {}",
+              numWarmingRequests,
+              numQueriesThreshold);
+          responseObserver.onError(
+              Status.UNKNOWN
+                  .withDescription(
+                      String.format(
+                          "Unable to backup warming queries since warmer has {} requests, which is less than threshold {}",
+                          numWarmingRequests,
+                          numQueriesThreshold))
+                  .asRuntimeException());
+          return;
+        }
+        int uptimeMinutesThreshold = request.getUptimeMinutesThreshold();
+        int currUptimeMinutes =
+            (int) (ManagementFactory.getRuntimeMXBean().getUptime() / 1000L / 60L);
+        if (uptimeMinutesThreshold > 0 && currUptimeMinutes < uptimeMinutesThreshold) {
+          logger.warn(
+              "Unable to backup warming queries since uptime is {} minutes, which is less than threshold {}",
+              currUptimeMinutes,
+              uptimeMinutesThreshold);
+          responseObserver.onError(
+              Status.UNKNOWN
+                  .withDescription(
+                      String.format(
+                          "Unable to backup warming queries since uptime is {} minutes, which is less than threshold {}",
```

Comment:
You need to provide `%s` as the the placeholder instead of `{}` in `String.format`